src/pull.go

```diff
@@ -82,6 +92,76 @@ func (g *Commands) Pull(byId bool) error {
 	return g.playPullChanges(nonConflicts, g.opts.Exports, opMap)
 }
 
+func autoRenameClashes(g *Commands, clashes []*Change) error {
+	type renameOp struct {
+		newName      string
+		change       *Change
+		originalPath string
+	}
+	clashesMap := map[string][]*Change{}
+
+	for _, clash := range clashes {
+		group := clashesMap[clash.Path]
+		if clash.Src == nil {
+			continue
+		}
+		clashesMap[clash.Path] = append(group, clash)
+	}
+
+	renames := make([]renameOp, 0, len(clashesMap)) // we will have at least len(clashesMap) renames
```

Comment:
Ah I see, you are right, I got that wrong. I was making the edits on my phone and hadn't noticed you specified the len as 0 then capacity as len(clashesMap) all I saw was make([]renameOp, len(clashesMap)). Sure then let's keep it as you have it.